src/lib.rs

```diff
@@ -319,13 +347,11 @@ where
         let node2str_id = |node: G::NodeId| -> String { format!("n{}", self.graph.to_index(node)) };
         // Emit an attribute for either node or edge
         // This will also keep track of updating the global attributes list
-        let mut emit_attribute = |writer: &mut EventWriter<_>,
+        let emit_attribute = |writer: &mut EventWriter<_>,
```

Comment:
This no longer needs to be a closure since no variables are  captured anymore. It would be nicer to just make this a normal function

```rust
fn emit_attribute<W>(writer: &mut EventWriter<W>, name: Cow<'static, str>, data: &str) -> WriterResult<()> where W: Write
```